### PR TITLE
[ISSUE #4593] Add server identity to replace user-agent white list.

### PR DESCRIPTION
**Pull request by KomachiSion**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

For #4593 , Add server identity to replace user-agent white list.

Templately Fix security problem. User can shutdown user-agent white list and defined their own identity key and value for request between servers.

## Brief changelog

- Add server identity to replace user-agent white list.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by KeRan213539 — APPROVED** [review]

File: core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java [hunk 1]
```diff
@@ -73,10 +73,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
...
+        if (authConfigs.isEnableUserAgentAuthWhite()) {
+            String userAgent = WebUtils.getUserAgent(req);
+            if (StringUtils.startsWith(userAgent, Constants.NACOS_SERVER_HEADER)) {
+                chain.doFilter(request, response);
+                return;
+            }
+        } else if (StringUtils.isNotBlank(authConfigs.getServerIdentityKey()) && StringUtils
+                .isNotBlank(authConfigs.getServerIdentityValue())) {
+            String serverIdentity = req.getHeader(authConfigs.getServerIdentityKey());
+            if (authConfigs.getServerIdentityValue().equals(serverIdentity)) {
```
Comment (QuantumGhost) [inline core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java:85]: Please note that comparing credential with `equals` makes your application vulnerable to [timing attack](https://en.wikipedia.org/wiki/Timing_attack)

Comment (KomachiSion) [reply core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java:85]: How to solve it? Can you provide some advise for this?

Comment (Slahser) [reply core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java:85]: https://github.com/playframework/play1/blob/master/framework/src/play/mvc/CookieDataCodec.java#L82

Comment (KomachiSion) [reply core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java:85]: In my opinion,  there is enough for equals. reason follow:

1. Nacos is Internal IDC application, we recommend users use Nacos with internal network not public network, If hacker want to timing attack, the network has been controlled by hacker. And Cost much time to analyze package between nacos server.
2. The key and value is defined by configuration file, as the first item said, if the internal network is in controlled by hacker. they can find the key and value by configuration file directly.

I'm not sure whether the timing attack is send a long string to hold equals resource. If so, I think we has no ability to solve all `public application security problem` for an internal network application.

Comment (Zheaoli) [reply core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java:85]: @KomachiSion FYI maybe you should use [`MessageDigest.isEquals` ](https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html)

**Pull request merged**